source/asm/TOOLCHAIN_GCC_ARM/CortexContextSwitch.s

```diff
@@ -0,0 +1,292 @@
+@ The MIT License (MIT)
```

Comment:
Is this file meant to replace https://github.com/lancaster-university/microbit-dal/blob/master/source/asm/CortexContextSwitch.s.gcc? If so should the old one be removed?

Reply:
Same as previous.

Reply:
So the makefile needs the `source/asm/CortexContextSwitch.s.gcc` as well?

Right now we have two GCC asm files:
- `source/asm/CortexContextSwitch.s.gcc`
- `source/asm/TOOLCHAIN_GCC_ARM/CortexContextSwitch.s`

So that it can copy one of them to `source/asm/CortexContextSwitch.s`?
How does it decide which one of the two to copy when using GCC?

Reply:
When using the makefile (yotta?) it ignored the `TOOLCHAIN_*` directory and checks the compiler flags to copy the correct `.s.gcc/armcc` file to `.s`.

Reply:
Where is this makefile or what tool/steps generates it?

Reply:
https://github.com/lancaster-university/microbit-dal/blob/master/source/CMakeLists.txt

Reply:
Thanks!
So it looks like we are manually maintaining that CMake file anyway, and in that case we can move the original GCC and ARMCC assembly files to the new location required by mbed-cli, and update these lines to find them:
https://github.com/lancaster-university/microbit-dal/blob/8c2403e9643f28d19ecd852ed9916fbc090496d2/source/CMakeLists.txt#L88-L94